core/Tracker/Visit/VisitProperties.php

```diff
@@ -52,27 +63,72 @@ public function &getProperties()
      *
      * @param string $name The property name.
      * @param mixed $value The property value.
+     *
+     * @return void
      */
-    public function setProperty($name, $value)
+    public function setProperty($name, $value): void
     {
         $this->visitInfo[$name] = $value;
     }
 
     /**
      * Unsets all visit properties.
+     *
+     * @return void
      */
-    public function clearProperties()
+    public function clearProperties(): void
     {
-        $this->visitInfo = array();
+        $this->visitInfo = [];
     }
 
     /**
      * Sets all visit properties.
      *
      * @param array $properties
+     *
+     * @return void
      */
-    public function setProperties($properties)
+    public function setProperties(array $properties): void
     {
         $this->visitInfo = $properties;
     }
+
+    /**
+     * Set the initial value of a property, this should remain unchanged throughout request processing.
+     *
+     * @param string $name
+     * @param mixed $value
+     *
+     * @return void
+     */
+    public function initializeImmutableProperty(string $name, $value): void
```

Comment:
I wonder if this methods makes sense from a logical point of view. The constructor method of this class sets both, the immutable property and the property that might be subject to change.
Wouldn't it then make more sense to only have one method like `initializeProperty`, which sets both? I can't see a logical reason why it would be needed or would make sense to set the immutable property only, but not the other one as well.

Reply:
That makes sense :+1: 

I've adjusted the methods:
```
initializeProperty()   // Sets both the mutable and immutable property values
setProperty()          // Sets just the mutable property value
getProperty()          // Get the mutable property value
getImmutableProperty() // Get the immutable property value
```
I also changed the wrapper method in the Visitor class to work the same way.